pkg/manager/member/pd_failover.go

```diff
@@ -210,26 +207,94 @@ func (f *pdFailover) tryToDeleteAFailureMember(tc *v1alpha1.TidbCluster) error {
+			klog.Infof("pd failover[tryToDeleteAFailureMember]: failure pod %s/%s has DeletionTimestamp = %s", ns, failurePodName, pod.DeletionTimestamp)
+			// Use force option to delete the pod only after all PVCs associated with the Pod are deleted
+			// If the hosting node is no more available or if the pod Ready condition has been False for a long time, then force delete the pod
+			if len(pvcs) == 0 && (hostingNodeUnavailable || podReadyCond.LastTransitionTime.Add(forceDeleteLongWait).After(time.Now())) {
+				if err := f.deps.PodControl.ForceDeletePod(tc, pod); err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		klog.Infof("pd failover[tryToDeleteAFailureMember]: failure pod %s/%s not found, skip", ns, failurePodName)
+	}
+
 	for _, pvc := range pvcs {
 		_, pvcUIDExist := failureMember.PVCUIDSet[pvc.GetUID()]
 		// for backward compatibility, if there exists failureMembers and user upgrades operator to newer version
 		// there will be failure member structures with PVCUID set from api server, we should handle this as pvcUIDExist == true
 		if pvc.GetUID() == failureMember.PVCUID {
 			pvcUIDExist = true
 		}
-		if pvc.DeletionTimestamp == nil && pvcUIDExist {
+		if !pvcUIDExist {
+			continue
+		}
+		if pvc.DeletionTimestamp == nil {
 			if err := f.deps.PVCControl.DeletePVC(tc, pvc); err != nil {
 				klog.Errorf("pd failover[tryToDeleteAFailureMember]: failed to delete PVC: %s/%s, error: %s", ns, pvc.Name, err)
 				return err
 			}
 			klog.Infof("pd failover[tryToDeleteAFailureMember]: delete PVC %s/%s successfully", ns, pvc.Name)
+		} else {
+			klog.Infof("pd failover[tryToDeleteAFailureMember]: DeletionTimestamp = %s set for failure PVC %s/%s", pvc.DeletionTimestamp, ns, pvc.Name)
+			// If the hosting node is no more available or if the pvc is stuck in deletion for a long time, then force delete the pvc
+			if hostingNodeUnavailable || pvc.DeletionTimestamp.Add(forceDeleteLongWait).After(time.Now()) {
+				klog.Infof("pd failover[tryToDeleteAFailureMember]: Finalizers=%v set for failure PVC %s/%s", pvc.Finalizers, ns, pvc.Name)
+				if removePvcDeleteProtectionFinalizer(pvc) {
+					if _, err := f.deps.PVCControl.UpdatePVC(tc, pvc); err != nil {
+						klog.Errorf("pd failover[tryToDeleteAFailureMember]: failed to remove finalizer from PVC: %s/%s, error: %s", ns, pvc.Name, err)
+						return err
+					}
+					klog.Infof("pd failover[tryToDeleteAFailureMember]: remove finalizer from PVC %s/%s successful", ns, pvc.Name)
+				}
+			}
 		}
 	}
 
 	setMemberDeleted(tc, failurePDName)
 	return nil
 }
 
+// isHostingNodeUnavailable checks whether the hosting node is unavailable
+// 1. Check the pod Status, if the pod has Unknown status phase, then it means node is not available
+// 2. Check the node Status, if the Ready condition of node is False, then it means node is not available
+func (f *pdFailover) isHostingNodeUnavailable(pod *apiv1.Pod) (bool, error) {
+	ns := pod.Namespace
+	name := strings.Split(pod.Name, ".")[0]
```

Comment:
why need this? why not origin pod name

Reply:
I have actually followed the same code that is present in tryToDeleteAFailureMember method where the the pod name string for display in logs is used as `failurePodName = strings.Split(pdName, ".")[0]`